### DISCUSSION TITLE
Subjective Feedback on Ajax Off-Canvas Cart/Basket in BootScore Version 6

### DISCUSSION THREAD

**Discussion by hsankala** [bootscore/bootscore#883] [General]
Yo Basti,  

I hope this finds you well! As I’m wrapping up my upgrade to version 6 of Boot Score for my eCommerce site, I wanted to open a discussion regarding some aspects of the Ajax off-canvas cart/basket menu. Here are my thoughts:  

### **1. Product Excerpt Display**  
- **Current Behavior**: The cart menu includes truncated product excerpts, which sometimes contain rendered HTML tags like `<tables>`.  
- **Concerns**:  
  - This makes the menu feel a bit busy, particularly with verbose descriptions or headings.  
  - The truncated text doesn’t seem to add much value in this context.  
- **Suggestion**: Introduce a **filter** to allow users to **disable the product excerpt** in the cart menu for a cleaner and simpler layout.  

---

### **2. Product Titles**  
- **Current Behavior**: Product titles are limited by a `text-truncate` CSS class.  
- **Issue**: This truncation can cut off important information, reducing the cart's usability.  
- **Observation**: In version 5, full product titles were displayed, which provided a better user experience.  
- **Suggestion**: Apply a **filter** or **add a CSS ID to the anchor tag** for the product name, so it can be easily targeted with CSS. This would allow users to override the text truncation CSS rules and display the full product title in the cart.

---

### **3. Stock Level Notifications**  
- **Current Behavior**: The "Only X left in stock" notification is always displayed in the cart menu.  
- **Concerns**: While helpful, this feature might not be desirable for all users.  
- **Suggestion**: Introduce a **filter** to **enable or disable the stock level notification** for more customization flexibility.  

---

### **Summary of Proposed Enhancements**  
To make the cart/basket menu more customisable, I suggest adding:  
1. **A filter to disable the product excerpt display.**  
2. **A filter to control the truncation of product titles.**  
3. **A filter to enable/disable the stock level notification.**  

I believe these changes would be useful for customising the flyout cart, offering more control over what is shown and how it is displayed, without requiring a full template overwrite. These suggestions are just subjective observations based on my experience and are not critical or essential by no means—I'm not pushing for them to be implemented, but I do think they could contribute to a cleaner, more streamlined menu. This is especially relevant for the site I’m working with, where a simpler design would enhance the user experience.

Here is an exmaple site cart of the website I'm doing at version 6

![image](https://github.com/user-attachments/assets/3832f04a-849f-4cc9-9c93-4a88d88478fb)

And here is the cart menu under Bootscore version 5.3 - I think this is tidier/less busy and fits more in. Less is More ;)

![image](https://github.com/user-attachments/assets/1169f18f-d3b9-4cf2-84f1-0a3c5824cb3d)

Thanks,  
Hefin

**Comment by justinkruit** [comment]
I must just say; excellent write-up! I also agree with these points. On point 2, isn't the title already selectable using CSS?

**Reply by hsankala** [reply]
Cheers. Na, it doesn't have a CSS ID selector on the anchor tag 

https://github.com/bootscore/bootscore/blob/4e37a62324f616850d53a5998d25a57743cd9d64/woocommerce/cart/mini-cart.php#L65-L68

**Reply by crftwrk** [reply]
Totally agree with that. On point 2, there is indeed no unique selector except this combined Bootstrap classes. Think an ID is not useful here, but a class like for example `cart-product-title` would be helpful. Something like this?

```php
class="cart-product-title <?= apply_filters('bootscore/class/cart/product-title', 'h6 text-decoration-none d-block text-truncate mb-0'); ?>"
```


@hsankala It would be great if you just open a PR to https://github.com/bootscore/bootscore/tree/Add-more-filters branch, so it will be part of https://github.com/bootscore/bootscore/pull/841.

Any thoughts?

**Reply by hsankala** [reply]
@crftwrk No worries I've open a PR for this on the Add-more-filters branch referencing this discussion/changes.

The issue I'm addressing with the product titles (point two) is primarily due to the `truncate-text` class.  Thinking about it ID selector is no good, as there could well be multiple Products in the AJAX cart and this would mean the same ID would be given to the product title tag multiple times on the page! (which isn't good HTML ;) 

So I’m happy to go with the 'cart-product-titles' class, but I'm open to any other ideas, ...the Bootstrap 5 text truncate class has been applied to the Cart product titles

```css
.text-truncate {
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
}
```
Having a specific CSS class would mean I could revert/nullify the text-truncate CSS rules with something like this in my own stylesheet 
```
/* Override text-truncate class when applied with another specific class */
.cart-product-titles.text-truncate {
    overflow: visible !important; /* Reset overflow */
    text-overflow: clip !important; /* Reset text overflow */
    white-space: normal !important; /* Reset white-space */
}
```
